### ISSUE TITLE
Add an isSupported method

### ISSUE THREAD

**Issue by wilsonge** [paragonie/sodium_compat#55]
Basically giving some sort of `isSupported` method for use in mass market things like Joomla. That basically means exposing some sort of form of the `use_fallback` method. https://github.com/paragonie/sodium_compat/blob/master/src/Compat.php#L2436

This is so we can have a comprehensive `isSupported` method https://github.com/joomla-framework/authentication/blob/2.0-dev/src/Password/Argon2iHandler.php#L73-L79 that allows us to either enable or disable Argon support depending on the users environment for mass market

(for example right now our travis fails because libsodium isn't installed which this library depends on - and i'd rather not recode up what's going on in the `use_fallback` method here https://travis-ci.org/joomla/joomla-cms/builds/308145677

**Comment by paragonie-scott** [comment]
This will be implemented in v1.4.0 as:

```php
var_dump(
    # Argon2:
    ParagonIE_Sodium_Compat::crypto_pwhash_is_available(),
    # Scrypt (legacy, but still bundled in ext/sodium):
    ParagonIE_Sodium_Compat::crypto_pwhash_scryptsalsa208sha256_is_available()
);
```

**Comment by wilsonge** [comment]
Awesome :) Thankyou!
